File: Python/Mahasiswa.py
# Saya Hakasa Putri mengerjakan LP3  DPBO 2023 C2 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
# kecurangan seperti yang telah dispesifikasikan. Aamiin 

#melakukan import class
from SivitasAkademik import SivitasAkademik
from Prodi import Prodi

class Mahasiswa(SivitasAkademik):

    # atribut private
    __NIM = ""
    __Fakultas = ""


    def __init__(self): #Constructor

        # atribut
        self. __NIM = ""
        self.__Fakultas = ""

    # setter getter
    def getNIM(self):
        return self.NIM
    
    def setNIM(self, NIM):
        self.NIM = NIM

    def getFakultas(self):
        return self.Fakultas
    
    def setFakultas(self, Fakultas):
        self.Fakultas = Fakultas

    def getProdi(self):
        return self.Prodi
    
    def setProdi(self, Prodi):
        self.Prodi = Prodi

    # untuk print atribut kelas mahasiswa
    def getInfoMahasiswa(self):

        print("-------------------------------------------")
        print("yang merupakan seorang mahasiswa dengan")
        print("-------------------------------------------")
        print("NIM            : " + self.getNIM())
        print("Fakultas       : " + self.getFakultas())
       
        self.Prodi.getInfoProdi()

File: C++/Program/Mahasiswa.cpp
// Saya Hakasa Putri mengerjakan LP3 DPBO 2023 C2
// dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan
// kecurangan seperti yang telah dispesifikasikan. Aamiin

#include <iostream>
// #include "SivitasAkademik.cpp" // Change file extension from .cpp to .h
// #include "Prodi.cpp" // Change file extension from .cpp to .h

class Mahasiswa{
private:
string NIM;
string Fakultas;
Prodi prodi; // Change variable name to lowercase
public:
// Constructor
Mahasiswa() {
NIM = "";
Fakultas = "";
}

// Getter dan setter
string getNIM() {
    return NIM;
}
void setNIM(string NIM) {
    this->NIM = NIM;
}

string getFakultas() {
    return Fakultas;
}
void setFakultas(string Fakultas) {
    this->Fakultas = Fakultas;
}

Prodi getProdi() {
    return prodi; // Change variable name to lowercase
}
void setProdi(Prodi prodi) { // Change variable name to lowercase
    this->prodi = prodi; // Change variable name to lowercase
}

// Print atribut kelas Mahasiswa
void getInfoMahasiswa() {
    cout << "-------------------------------------------" << endl;
    cout << "yang merupakan seorang mahasiswa dengan" << endl;
    cout << "-------------------------------------------" << endl;
    cout << "NIM            : " << getNIM() << endl;
    cout << "Fakultas       : " << getFakultas() << endl;

    prodi.getInfoProdi(); // Change variable name to lowercase
}

};